SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialSender.cpp

```diff
@@ -209,9 +209,14 @@ void USpatialSender::SendAuthorityIntentUpdate(const AActor& Actor, VirtualWorke
 
 	AuthorityIntent* AuthorityIntentComponent = StaticComponentView->GetComponentData<AuthorityIntent>(EntityId);
 	check(AuthorityIntentComponent != nullptr);
-	checkf(AuthorityIntentComponent->VirtualWorkerId != NewAuthoritativeVirtualWorkerId,
-		   TEXT("Attempted to update AuthorityIntent twice to the same value. Actor: %s. Entity ID: %lld. Virtual worker: '%d'"),
-		   *GetNameSafe(&Actor), EntityId, NewAuthoritativeVirtualWorkerId);
+	if (AuthorityIntentComponent->VirtualWorkerId == NewAuthoritativeVirtualWorkerId)
+	{
+		/* It's possible for ReplicationGraph to behave incorrectly and trigger an authority change more than once. Please see
```

Comment:
`Please see SpatialReplicationGraph for reference.` 
What reference can they see in the rep graph?

Reply:
Hmm I think mainly as an example, since their conditions for replications don't specifically match the expectations this check assumes. Maybe the comment is misleading though

Reply:
I think because this is driven by our code, it's not super useful to point users at the replication graph. Perhaps something more like, "this seems to occur when using the replication graph, however we're still unsure the cause".

Also do we want to early out from this function when we hit this? Seems reasonable. @alastairdglennie did you do that in your workaround?

Reply:
Yeah we can early out, I was unsure which was more appropriate because the state is invalid and there didn't seem to be any harm sending the intent again.

Ally said in the initial ticket that they did early out so maybe that is the better option since he had context from Intrepid

Reply:
Early out was the workaround, yep. Ticket was raised because while setting AuthorityIntent might be a case we can ignore, Actors are twice in same tick seems like unexpected state that could've caused other issues.